Sources/Core/EthereumNetwork/Request/APIRequest+Methods.swift

```diff
@@ -36,7 +36,7 @@ extension APIRequest {
 
         /// This bit of code is purposed to work with literal types that comes in ``Response`` in hexString type.
         /// Currently it's just `Data` and any kind of Integers `(U)Int`, `Big(U)Int`.
-        if Result.self == Data.self || Result.self == UInt.self || Result.self == Int.self || Result.self == BigInt.self || Result.self == BigUInt.self {
+        if Result.self == Data.self || Result.self == UInt.self || Result.self == Int.self || Result.self == BigInt.self || Result.self == BigUInt.self || Result.self == String.self {
```

Comment:
you should also respect `String: LiteralInitiableFromString`. 

```swift
extension String: LiteralInitiableFromString {
    init?(from hexString: String) {
        self.init(hexString)
    }
}
```

Reply:
This one is for decoding response from a node, not encoding a request. So this does not affects encoding data to hexstring in anyway.

Reply:
This actual change was not tested and thought through completely.
After reading the implementation of this parsing once more I see that this change is most likely redundant and will be reverted.
The last line of this function must cover the case when the generic `Result` type is `String`.